src/test/groovy/net/vivin/gradle/versioning/TestRepository.groovy

```diff
@@ -130,4 +135,15 @@ class TestRepository {
             .setTarget(repository.resolve(Constants.HEAD))
             .call()
     }
+
+    String getHeadTagMessage() {
+        def revWalk = new RevWalk(repository)
+        def head = repository.resolve(Constants.HEAD)
+
+        return repository.tags.values()
+            .collect { repository.peel(it) }
+            .findAll { it.peeledObjectId == head }
+            .collect { revWalk.parseTag(it.objectId).fullMessage }
+            .find()
+    }
```

Comment:
My reason was that I couldn't figure out how to get this to work at all :D and plus I was doing this really late at night and I was getting annoyed already haha. I am sure your method is nicer :)